items/resources.py

```diff
@@ -132,6 +159,32 @@ class Meta:
       'deadline': ('gt', ),
     }
 
+class FileResource(ModelResource):
+  group = fields.ToOneField(GroupResource, 'group')
+  class Meta:
+    queryset = File.objects.all()
+    resource_name = 'file'
+    authentication = SessionAuthentication()
+    authorization = FileAuthorization()
+
+  def prepend_urls(self):
+    return [
+      url(r'^%s/upload/$' % self._meta.resource_name, self.wrap_view('upload'), name='api_file_upload'),
+    ]
+
+  def upload(self, request, **kwargs):
+    self.method_check(request, allowed=['post'])
+    if self.is_authenticated(request):
```

Comment:
it's not work in this way. Just call `self.is_authenticated(request)` , it will generate Unathorized response if needed